libraries/abstractions/secure_sockets/test/iot_test_tcp.c

```diff
@@ -1544,11 +1544,17 @@ static void prvSOCKETS_NonBlocking_Test( Server_t xConn )
                 xNumBytesReceived += lNumBytes;
             }
 
+            if( lNumBytes < 0 )
+            {
+                break;
+            }
+
             xEndTime = xTaskGetTickCount();
         }
-        while( ( ( xEndTime - xStartTime ) < xWaitTime ) && ( xMessageLength > xNumBytesReceived ) );
+        while( xMessageLength > xNumBytesReceived );
 
-        TEST_ASSERT_EQUAL_INT32_MESSAGE( xMessageLength, xNumBytesReceived, "Data was not received \r\n" );
+        xResult = prvCheckTimeout( xStartTime, xEndTime, xTimeout );
+        TEST_ASSERT_EQUAL_INT32_MESSAGE( pdPASS, xResult, "Receive timeout was outside of acceptable range" );
```

Comment:
I think using `TEST_ASSERT_EQUAL_MESSAGE` would be a better choice here because `xResult` is of type `BaseType_t`.

Reply:
Is this check no longer necessary?
```
TEST_ASSERT_EQUAL_INT32_MESSAGE( xMessageLength, xNumBytesReceived, "Data was not received \r\n" );
```

Reply:
Also, everytime we call `prvCheckTimeout`, we end up calling:
```
TEST_ASSERT_EQUAL_INT32_MESSAGE( pdPASS, xResult, "Receive timeout was outside of acceptable range" );
```
In that case, would it be better to just remove this check inside `prvCheckTimeout`?

Reply:
Thanks for the suggestion, fixed in ba515b4.
Starting a new CI tests will wait for the results and resolve this comment if it gets ideal result.